prow/cmd/generic-autobumper/main.go

```diff
@@ -19,47 +19,467 @@ package main
 import (
 	"fmt"
 	"io/ioutil"
+	"net/http"
+	"os"
+	"path/filepath"
+	"regexp"
+	"sort"
+	"strings"
 
 	flag "github.com/spf13/pflag"
 
 	"github.com/sirupsen/logrus"
 
+	imagebumper "k8s.io/test-infra/experiment/image-bumper/bumper"
 	"k8s.io/test-infra/prow/cmd/generic-autobumper/bumper"
 
 	"sigs.k8s.io/yaml"
 )
 
-func parseOptions() (*bumper.Options, error) {
+const (
+	latestVersion          = "latest"
+	upstreamVersion        = "upstream"
+	upstreamStagingVersion = "upstream-staging"
+	tagVersion             = "vYYYYMMDD-deadbeef"
+	defaultUpstreamURLBase = "https://raw.githubusercontent.com/kubernetes/test-infra/master"
+)
+
+var (
+	tagRegexp    = regexp.MustCompile("v[0-9]{8}-[a-f0-9]{6,9}")
+	imageMatcher = regexp.MustCompile(`(?s)^.+image:(.+):(v[a-zA-Z0-9_.-]+)`)
+)
+
+var _ bumper.PRHandler = (*client)(nil)
+
+type client struct {
+	o        *Options
+	images   map[string]string
+	versions map[string][]string
+}
+
+// Changes returns a slice of functions, each one does some stuff, and
+// returns commit message for the changes
+func (c *client) Changes() []func() (string, error) {
+	return []func() (string, error){
+		func() (string, error) {
+			var err error
+			c.images, err = UpdateReferences(c.o)
+			if err != nil {
```

Comment:
I'd combine with the previous line, since you're using using `=` (applies everywhere, think I already wrote this somewhere else once)

Reply:
done